src/switch_caller.c

```diff
@@ -587,7 +585,92 @@ SWITCH_DECLARE(void) switch_caller_extension_add_application(switch_core_session
 		caller_extension->last_application = caller_application;
 		caller_extension->current_application = caller_extension->applications;
 	}
+}
 
+SWITCH_DECLARE(void) switch_caller_extension_insert_applications(switch_core_session_t *session,
+																switch_caller_extension_t *caller_extension,
+																const char **application_names,
+																const char **application_data,
+																size_t app_count,
+																char **low_priority_applications,
+																size_t low_priority_count)
+{
+    switch_caller_application_t *new_applications = NULL, *last_new_application = NULL;
+	switch_caller_application_t *prev = NULL, *current;
+
+	switch_assert(session != NULL);
+    switch_assert(application_names != NULL);
+    switch_assert(application_data != NULL);
+
+    // Allocate memory for all applications upfront
+    for (size_t i = 0; i < app_count; ++i) {
+    	switch_caller_application_t *caller_application;
+
+    	if (!application_names[i] || !application_data[i]) {
```

Comment:
- Check both NULL and Empty for names.
- Some application can have empty or null data e.g ring_ready

Reply:
done